astropy/coordinates/representation.py

```diff
@@ -223,6 +224,27 @@ def _unitstr(self):
                            for component in self.components]))
         return unitstr
 
+    def __eq__(self, other):
+        """
+        Check for equality with another representation.
+        """
+        try:
+            comparisons = [getattr(self, component) ==
+                           getattr(other, component)
+                           for component in self.components]
+        except:
+            try:
+                newrep = other.represent_as(self.__class__)
+                comparisons = [getattr(self, component) ==
+                               getattr(newrep, component)
+                               for component in self.components]
+            except:
+                return False
+        return functools.reduce(operator.and_, comparisons)
```

Comment:
What about `all()` here? `operator.and_` is the bitwise and operator and should be equivalent to `all`?